### PR TITLE
Check Hamiltonians are hermitian

### PR DESCRIPTION
**Pull request by JohanSchott**
Add check that non-interacting and full Hamiltonians are hermitian.

### CODE REVIEW

File: impurityModel/ed/finite.py [hunk 1]
```diff
@@ -287,6 +287,12 @@ def daggerOp(op):
     return opDagger
 
 
+def check_hermitian(op: dict[tuple, int|float|complex]):
+    for process, value in daggerOp(op).items():
+        assert process in op
```
Comment (kalvdans) [inline impurityModel/ed/finite.py:292]: a bit unnecessary as op[process] will raise KeyError anyway on the next line.

Comment (JohanSchott) [reply impurityModel/ed/finite.py:292]: yes. 
I thought maybe it was easier to debug when checks were split up.
But let's remove this line, KeyError is pretty clear.

Comment (JohanSchott) [reply impurityModel/ed/finite.py]: Safe to compare the two dicts with == ? 

Pushed this suggestion.

Comment (kalvdans) [reply impurityModel/ed/finite.py]: Yes, it's safe!

much neater.

It's always a red flag to name a function check_something as it isn't clear from the name what the return value is. Better call it `assert_hermetian`.

Comment (JohanSchott) [reply impurityModel/ed/finite.py]: renamed function

**Review by kalvdans — APPROVED** [review]

**Pull request merged**